test/metabase/models/dashboard_test.clj

```diff
@@ -243,20 +291,229 @@
+                                                                  :row              0
+                                                                  :col              0
+                                                                  :size_x           4
+                                                                  :size_y           4}
+                                                                 {:dashboard_id     dashboard-id
+                                                                  :dashboard_tab_id tab-2-id
+                                                                  :row              4
+                                                                  :col              4
+                                                                  :size_x           4
+                                                                  :size_y           4}])]
+      (create-dashboard-revision! dashboard-id false)
+
+      ;; 2.a: tab 1: delete the 2nd card, add 2 cards and update position of one card,
+      (t2/insert! :model/DashboardCard [{:dashboard_id     dashboard-id
+                                         :dashboard_tab_id tab-1-id
+                                         :row              0
+                                         :col              0
+                                         :size_x           4
+                                         :size_y           4}
+                                        {:dashboard_id     dashboard-id
+                                         :dashboard_tab_id tab-1-id
+                                         :row              0
+                                         :col              0
+                                         :size_x           4
+                                         :size_y           4}])
+      (t2/delete! :model/DashboardCard card-2-tab-1)
+      (t2/update! :model/DashboardCard card-1-tab-1 {:row 10 :col 10})
+
+      ;; 2.b: delete tab 2
+      (t2/delete! :model/DashboardTab tab-2-id)
+
+      ;; 2.c: create a new tab with 1 card
+      (let [new-tab-id (t2/insert-returning-pks! :model/DashboardTab {:name         "Tab 3"
+                                                                      :position     1
+                                                                      :dashboard_id dashboard-id})]
+        (t2/insert! :model/DashboardCard {:dashboard_id     dashboard-id
+                                          :dashboard_tab_id new-tab-id
+                                          :row              0
+                                          :col              0
+                                          :size_x           4
+                                          :size_y           4}))
+      (create-dashboard-revision! dashboard-id false)
+
+     ;; revert
+     (revert-to-previous-revision Dashboard dashboard-id 2)
+     (testing "should works"
```

Comment:
This just adds noise to the test description imo
```suggestion
```

Reply:
removed  in c7a6e58f97